lib/seahorn/BvOpSem2.cc

```diff
@@ -914,12 +940,19 @@ class OpSemMemManager {
     for (unsigned i = 0; i < len; i += sem_word_sz) {
       Expr offset = bv::bvnum(i, ptrSzInBits(), m_efac);
       Expr dIdx = i == 0 ? dPtr : mk<BADD>(dPtr, offset);
-      unsigned word = 0;
+      // copy bytes from buffer to word - word must accommodate largest supported word size
+      unsigned long word = 0;  // 8 bytes because assumed largest supported sem_word_sz = 8
```

Comment:
Add an assertion/warning when the assumption is violated. We might have very large semanitc words in the future. It is fine not to support that case now, but the code should be defensive against such future change.